BankChain/app/src/androidTest/java/nl/tudelft/ewi/ds/bankchain/bank/bunq/BunqTest.java

```diff
@@ -144,4 +153,100 @@ public void testExceptionConfirmationNotFuture() {
+        bank.createSession().get();
+        users = bank.listUsers().get();
+        Party firstUser = users.get(0);
+        assertEquals(firstUser.getId(), 2002);
+        assertEquals(firstUser.getName(), "Overton Onderlinge Waarborgmaatschappij");
+    }
+
+
+    @Test
+    public void testGetAccounts() throws ExecutionException, InterruptedException {
+        CompletableFuture<Void> future;
+        Bank bank;
+        Environment env;
+
+        env = new Environment();
+        env.setBank("Bunq");
+
+        env.setBankUrl("REAL_BUNQ_URL");
+        env.setBankApiKey("REAL_BUNQ_API_KEY");
+
+        bank = new BankFactory(env).create();
+        bank.createSession().get();
+
+        List<Account> accounts = bank.listAccount(new BunqParty("Overton Onderlinge Waarborgmaatschappij", 2002)).get();
+        Account ac = accounts.get(0);
+        assertEquals(ac.getParty().getId(), 2002);
+        assertEquals(ac.getIban(), "NL05BUNQ9900019989" +
+                "");
+        assertEquals(ac.getId(), 2021);
+    }
+
+    @Test
+    public void testListTransactions() throws ExecutionException, InterruptedException {
+        CompletableFuture<Void> future;
+        Bank bank;
+        Environment env;
+
+        env = new Environment();
+        env.setBank("Bunq");
+
+        env.setBankUrl("REAL_BUNQ_URL");
+        env.setBankApiKey("REAL_BUNQ_API_KEY");
+
+        bank = new BankFactory(env).create();
+        bank.createSession().get();
+
+        BunqParty party = new BunqParty("Overton Onderlinge Waarborgmaatschappij", 2002);
+        BunqAccount account = new BunqAccount("NL05BUNQ9900019989", 2021, party);
+
+        List<Transaction> transactions = bank.listTransactions(account).get();
+        Transaction t = transactions.get(transactions.size() - 2);
+        assertEquals(t.getId(), 14191);
+        assertEquals(t.getDescription(), "invoice 688");
+        assertEquals(t.getAcount().getId(), 2021);
+        assertEquals(t.getCounterAccount().getIban(), "NL28BUNQ9900000951");
+    }
+
+    @Test
+    public void testPostTransaction() throws ExecutionException, InterruptedException {
+        CompletableFuture<Void> future;
+        Bank bank;
+        Environment env;
+
+        env = new Environment();
+        env.setBank("Bunq");
+
+        env.setBankUrl("REAL_BUNQ_URL");
+        env.setBankApiKey("REAL_BUNQ_API_KEY");
+
+        bank = new BankFactory(env).create();
+        bank.createSession().get();
+
+        BunqParty party = new BunqParty("Overton Onderlinge Waarborgmaatschappij", 2002);
+        BunqAccount account = new BunqAccount("NL05BUNQ9900019989", 2021, party);
+
+        BunqParty cp = new BunqParty("Branda Monoghan", -1);
+        BunqAccount ca = new BunqAccount("NL77BUNQ9900016947", -1, cp);
+        BunqTransaction pay = new BunqTransaction(-0.01f, account, ca, Currency.getInstance("EUR"), "end tot end Bunqtest");
+
+        assertTrue(bank.sendTransaction(pay).get());
```

Comment:
Please also test that the balances are changed, the transaction shows up when you get the transaction list, etc...